Userland/Libraries/LibVT/Line.cpp

```diff
@@ -10,19 +10,86 @@ namespace VT {
 
 Line::Line(size_t length)
 {
-    set_length(length);
+    set_length(length, nullptr, nullptr);
 }
 
 Line::~Line()
 {
 }
 
-void Line::set_length(size_t new_length)
+void Line::set_length(size_t new_length, Line* next_line, CursorPosition* cursor, bool cursor_is_on_next_line)
 {
     size_t old_length = length();
     if (old_length == new_length)
         return;
-    m_cells.resize(new_length);
+
+    // Drop the empty cells
+    if (m_terminated_at.has_value())
+        m_cells.remove(m_terminated_at.value(), m_cells.size() - m_terminated_at.value());
+
+    if (next_line) {
+        if (old_length < new_length) {
+            // Take as many cells as would fit from the next line
+            if (!m_terminated_at.has_value()) {
+                auto cells_to_grab_from_next_line = min(new_length - old_length, next_line->length());
+                auto clear_next_line = false;
+                if (next_line->m_terminated_at.has_value()) {
+                    cells_to_grab_from_next_line = min(cells_to_grab_from_next_line, static_cast<size_t>(next_line->m_terminated_at.value()));
+                    if (cells_to_grab_from_next_line == *next_line->m_terminated_at) {
+                        m_terminated_at = old_length + *next_line->m_terminated_at;
+                        next_line->m_terminated_at.clear();
+                        clear_next_line = true;
+                    } else {
```

Comment:
I feel like this code is nested way more deeply than it needs to be.
Can we reorganize it a little bit and cut the nesting in ~half? :)

Reply:
I can try, but it's not immediately obvious to me how I would do that :thinking:

Reply:
I hope this nesting reduction is enough :P

Reply:
Thank you! Yes, I find it much more understandable now :)